core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -303,7 +287,294 @@ public static Properties filterAndRemovePrefix(Map<String, String> properties, S
+  }
+
+  static String renameTableSql() {
+    return renameSql(true);
+  }
+
+  static String renameViewSql() {
+    return renameSql(false);
+  }
+
+  private static String dropSql(boolean isTable) {
+    String tableName = isTable ? CATALOG_TABLE_NAME : CATALOG_VIEW_NAME;
+    String namespace = isTable ? TABLE_NAMESPACE : VIEW_NAMESPACE;
+    String name = isTable ? TABLE_NAME : VIEW_NAME;
+    return String.format(DROP, tableName, namespace, name);
+  }
+
+  static String dropTableSql() {
+    return dropSql(true);
+  }
+
+  static String dropViewSql() {
+    return dropSql(false);
+  }
+
+  private static String doCommitCreateSql(boolean isTable) {
+    String tableName = isTable ? CATALOG_TABLE_NAME : CATALOG_VIEW_NAME;
+    String namespace = isTable ? TABLE_NAMESPACE : VIEW_NAMESPACE;
+    String name = isTable ? TABLE_NAME : VIEW_NAME;
+    return String.format(DO_COMMIT_CREATE, tableName, namespace, name);
+  }
+
+  private static int update(
+      boolean isTable,
+      JdbcClientPool connections,
+      String catalogName,
+      TableIdentifier identifier,
+      String newMetadataLocation,
+      String oldMetadataLocation)
+      throws SQLException, InterruptedException {
+    return connections.run(
+        conn -> {
+          try (PreparedStatement sql = conn.prepareStatement(doCommitSql(isTable))) {
+            // UPDATE
+            sql.setString(1, newMetadataLocation);
+            sql.setString(2, oldMetadataLocation);
+            // WHERE
+            sql.setString(3, catalogName);
+            sql.setString(4, namespaceToString(identifier.namespace()));
+            sql.setString(5, identifier.name());
+            sql.setString(6, oldMetadataLocation);
+            return sql.executeUpdate();
+          }
+        });
+  }
+
+  static int updateTable(
+      JdbcClientPool connections,
+      String catalogName,
+      TableIdentifier tableIdentifier,
+      String newMetadataLocation,
+      String oldMetadataLocation)
+      throws SQLException, InterruptedException {
+    return update(
+        true, connections, catalogName, tableIdentifier, newMetadataLocation, oldMetadataLocation);
+  }
+
+  static int updateView(
+      JdbcClientPool connections,
+      String catalogName,
+      TableIdentifier viewIdentifier,
+      String newMetadataLocation,
+      String oldMetadataLocation)
+      throws SQLException, InterruptedException {
+    return update(
+        false, connections, catalogName, viewIdentifier, newMetadataLocation, oldMetadataLocation);
+  }
+
+  private static Map<String, String> getContent(
+      boolean isTable, JdbcClientPool connections, String catalogName, TableIdentifier identifier)
+      throws SQLException, InterruptedException {
+    return connections.run(
+        conn -> {
+          Map<String, String> tableOrView = Maps.newHashMap();
+
+          String getTableOrViewSqlStatement = JdbcUtil.getSql(isTable);
```

Comment:
this one can probably be just inlined: `conn.prepareStatement(JdbcUtil.getSql(isTable))`